src/Tests/TestData/UnusedPackageReferenceDeferred/Library/Library.csproj

```diff
@@ -0,0 +1,23 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Library</OutputType>
+    <TargetFramework>net472</TargetFramework>
+    <EnableReferenceTrimmer>false</EnableReferenceTrimmer>
+    <CollectDeclaredReferencesDependsOn>$(CollectDeclaredReferencesDependsOn);SetEnableReferenceTrimmer</CollectDeclaredReferencesDependsOn>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Newtonsoft.Json" Version="13.0.2" />
+  </ItemGroup>
+
+  <Target Name="SetEnableReferenceTrimmer">
```

Comment:
Does this not work? `<Target Name="SetEnableReferenceTrimmer" BeforeTargets="CollectDeclaredReferences">`

Reply:
Or hook to `PrepareForBuild`

Reply:
Yes, that works